packages/app/src/components/VisualizationOptions/Options/SeriesTable.js

```diff
@@ -62,11 +64,19 @@ const SeriesTable = ({
     }, [])
 
     const onItemChange = (changedItem, value, prop) => {
-        const series = [...optionItems]
-        const item = series.find(
+        const series = cloneDeep(optionItems)
+
+        const itemIndex = series.findIndex(
             item => item.dimensionItem == changedItem.dimensionItem
         )
-        item[prop] = value
+
+        if (prop === TYPE_PROP && value === visType) {
+            const { [prop]: remove, ...rest } = series[itemIndex]
+            series[itemIndex] = { ...rest }
+        } else {
+            series[itemIndex][prop] = value
+        }
+
         onChange(series)
```

Comment:
@martinkrulltott To me this kind of work is typically the reducer's responsibility, so to avoid doing all this logic at (a dumb?) component level, including deep cloning which is a bit nasty, maybe create a new action for updating a series object?

Reply:
(Instead of dispatching the full new series array)

Reply:
Good points, fixed, please see my latest commit 👍